base/boot.jl

```diff
@@ -52,8 +52,23 @@
 #abstract type AbstractArray{T,N} end
 #abstract type DenseArray{T,N} <: AbstractArray{T,N} end
 
+#struct GenericMemory{isatomic, T}
+#   length::Int
+#   const data::Ptr{Cvoid}
+#   Union{ # hidden data
+#       elements :: NTuple{length, T}
+#       owner :: Any
+#   }
+#end
+
+# struct MemoryRef{isatomic, T}
+#    mem::Memory{isatomic, T}
+#    data::Ptr{Cvoid}
+#end
+
 #mutable struct Array{T,N} <: DenseArray{T,N}
```

Comment:
Any idea how many places we're still relying on Array being defined in /src? Would it be possible to convert all those uses to svec or something?

Reply:
unfortunately way too many. Specifically IR and AST needs to be manipulated (both grown and shrunk)